hack/generator/pkg/codegen/pipeline_remove_empty_object_types.go

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package codegen
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+)
+
+// removeEmptyObjectTypes removes empty ObjectTypes and references to them
+func removeEmptyObjectTypes() PipelineStage {
```

Comment:
Maybe move this to separate PR that merges first?

Reply:
Not doing this, instead rolling it into the embedded resource removal stage (since it doesn't really do anything except in the context of that stage)